lib/application.js

```diff
@@ -423,7 +423,7 @@ function dataSourcesFromConfig(name, config, connectorRegistry, registry) {
         config.connector = require(connectorPath);
       }
     }
-    if (!config.connector.name)
+    if (typeof config.connector !== 'string' && !config.connector.name)
```

Comment:
hey @loay ,
`if (!config.connector)` won't work alone, as config.connector is indeed initialized to, say, `mongodb`, or `mysql`, etc. if datasources.json is correctly set up.

But still I can tinker by adding turning the test this way:
`if (config.connector && typeof config.connector.name === 'object')`

of course comparing object type is not 100% safe (arrays, dates, null which is already checked here), but i think it's robust enough given the surrounding code